sourcegraph/sourcegraph
# [SG-37914] Extract bootstrap style import `buttons`

## Description
Extract / Remove usage of bootstrap's style sheet import on codebase.

## Success criteria
- Bootstrap button import is removed from `client/branded/src/global-styles/index.scss`
- Migrate remaining usage of `btn`, `btn-*` classes to wildcard component.

## Implementation Detail

- [x] Migrate `.btn-icon` usage to `<Button variant="icon">` 
- [x] Migrate `.btn-link` usage to `<ButtonLink />`, if it's not a good fit, migrate to `<Button variant="link" />`
- [x] Migrate `.btn-block` usage to `<Button display="block" />`
- [x] Migrate remaining usage of `<button ...>` on codebase to wildcard `<Button />` component.
- [x] Remove usage of `@import 'bootstrap/scss/buttons'`
- [x] Verify there are no UI diff as a result of any of the steps above.

### Refs
[Gitstart ticket](https://app.gitstart.com/clients/sourcegraph/tickets/SG-37914)
[SG Issue](https://github.com/sourcegraph/sourcegraph/issues/37914)

## Test Plan
- Identify migrated usage of bootstrap class.
- Verify that they have no UI diff as a result of migrations carried out in this PR.
   - To test ```client/web/src/org/members/AddMemberToOrgModal.tsx```   and ```client/web/src/org/members/InviteMemberModal.tsx```
      - open ```client/web/src/org/area/routes.tsx``` and comment out ```Line 25  -  the condition for settings/members``` 
      - go to ```/organizations/sourcegraph/settings/members``` on the web app.

## Links
- [UserCodeHosts.tsx](http://localhost:3080/welcome)
- TourTask.tsx in storybook
- [HomeTab.tsx](http://localhost:3080/github.com/sourcegraph/sourcegraph)

## App preview:

- [Web](https://sg-web-contractors-sg-37914-1.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-uyfuoykhyb.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
